debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -206,11 +208,19 @@ private void captureSessionMemoryStatistics(OracleConnection connection) throws
     private LogMinerEventProcessor createProcessor(ChangeEventSourceContext context,
                                                    OraclePartition partition,
                                                    OracleOffsetContext offsetContext) {
-        if (OracleConnectorConfig.LogMiningBufferType.INFINISPAN.equals(connectorConfig.getLogMiningBufferType())) {
-            return new InfinispanLogMinerEventProcessor(context, connectorConfig, jdbcConnection, dispatcher,
+        final LogMiningBufferType bufferType = connectorConfig.getLogMiningBufferType();
+        if (bufferType.isInfinispanEmbedded()) {
```

Comment:
How about having methods like this on the enum itself rather then the if-then-else cascade here? You can have specific method implementations for each of the enum constants.